Reqnroll/CucumberMessages/PubSub/MessagesFormatterPlugin.cs

```diff
@@ -0,0 +1,53 @@
+﻿#nullable enable
+
+using Reqnroll.Plugins;
+using Reqnroll.UnitTestProvider;
+using Reqnroll.Events;
+using Reqnroll.Tracing;
+using Reqnroll.BoDi;
+using System;
+using System.Threading.Tasks;
+using System.IO;
+using System.Linq;
+using Reqnroll.CucumberMessages.Configuration;
+using Reqnroll.CucumberMessages.PayloadProcessing;
+using System.Text;
+using System.Collections.Concurrent;
+using Io.Cucumber.Messages.Types;
+using Reqnroll.Utils;
+
+
+namespace Reqnroll.CucumberMessages.PubSub
+{
+    /// <summary>
+    /// The MessagesFormatterPlugin is the subscriber to the CucumberMessageBroker. 
+    /// It receives Cucumber Messages and writes them to a file.
+    /// 
+    /// </summary>
+    public class MessagesFormatterPlugin : FileWritingFormatterPluginBase
+    {
+
+        public MessagesFormatterPlugin(ICucumberMessagesConfiguration configuration, IFileSystem fileSystem) : base(configuration, "messages", ".ndjson", "reqnroll_report.ndjson", fileSystem)
+        {
+        }
+
+        internal override void ConsumeAndWriteToFilesBackgroundTask(string outputPath)
```

Comment:
Could we make this method Async (File I/O)?

Reply:
I can work on that, but will require that I modify and submit a PR to the Cucumber team for the `Cucumber.HtmlFormatter` nuget package to make it async as well.
The `MessageFormattingPlugin` can be made async without delay (no 3rd party dependencies), but the `HtmlFormatter` is another thing.

And just to confirm, making the `ConsumeAndWriteToFileBackgroundTask` async still has value despite the fact that it runs on a background thread, right?

Reply:
OK. Working on it. The PR for the HtmlFormatter has been submitted to add async support to that. When that gets released to nuget, I will migrate this to async as well.

Reply:
@obligaron , please take another look at the formatter base class structure. I've separated out the file-based responsibilities into another class. The class hierarchy is now `MessagesFormatterPlugin` -> derives from `FileWritingFormatterPluginBase` -> derives from `FormatterPluginBase`. 
The `FormatterPluginBase` (lowest level) takes care of launching the (abstract) task and shutting it down. The `FileWritingFormatterPluginBase` takes care of file open/close, etc. 
The lowest level base class can now be used as the basis for formatters that do not do file IO operations (such as pushing Messages to a HTTP endpoint).